### PR TITLE
go docs

### PR DESCRIPTION
**Pull request by spinales**
Go docs

### CODE REVIEW

**Comment by spinales** [comment]
#4

File: extensions/exchanges/gdax/exchange.go [hunk 1]
```diff
@@ -21,6 +22,7 @@ type gdaxExchange struct {
 	BackfillRateLimit int64
 }
 
+// NewGDAXExchange add new exchange to coinbase(gdax)
```
Comment (cpurta) [inline extensions/exchanges/gdax/exchange.go]: We'll probably want to re-phrase this to "returns a new Coinbase (gdax) Exchange interface implementation"

File: extensions/exchanges/gdax/exchange.go [hunk 2]
```diff
@@ -73,6 +76,7 @@ func (exchange *gdaxExchange) GetTrades(productID string) ([]*model.Trade, error
 	return trades, nil
 }
 
+// GetBalance return account balance on coinbase(gdax)
```
Comment (cpurta) [inline extensions/exchanges/gdax/exchange.go]: "returns the current account balance held on Coinbase (gdax)"

File: extensions/exchanges/gdax/exchange.go [hunk 3]
```diff
@@ -42,6 +44,7 @@ func NewGDAXExchange(key, passphrase, secret string, httpClient *http.Client) *g
 	}
 }
 
+// GetTrades  returns coinbase(gdax) transaction history
```
Comment (cpurta) [inline extensions/exchanges/gdax/exchange.go]: "returns the transaction history for a specific product on the Coinbase (gdax) exchange"

File: extensions/exchanges/gdax/exchange.go [hunk 4]
```diff
@@ -99,6 +103,7 @@ func (exchange *gdaxExchange) GetBalance(currency string, asset string) (*model.
 	return balance, nil
 }
 
+// GetQuote returns the price for the exchange made
```
Comment (cpurta) [inline extensions/exchanges/gdax/exchange.go]: "returns the current quote price of a product on Coinbase (gdax)"

File: internal/cassandra/cassandra_client.go [hunk 5]
```diff
@@ -9,16 +9,19 @@ import (
 
 var _ Client = &cassandraClient{}
 
+// CassandraClient model for transactions with cassandra
```
Comment (cpurta) [inline internal/cassandra/cassandra_client.go]: Not sure if we really need to have documentation on private level structs. I think that we can remove this comment

File: extensions/exchanges/gdax/exchange.go [hunk 6]
```diff
@@ -12,6 +12,7 @@ var (
 	_ model.Exchange = &gdaxExchange{}
 )
 
+// gdaxExchange implements the representation of a coinbase(gdax) exchange
```
Comment (cpurta) [inline extensions/exchanges/gdax/exchange.go]: Not sure if we really need to have documentation on private level structs. I think that we can remove this comment

File: internal/cassandra/cassandra_client.go [hunk 7]
```diff
@@ -9,16 +9,19 @@ import (
...
 type cassandraClient struct {
 	session *gocql.Session
 }
 
+// NewCassandraClient return new instance of cassandraClient
```
Comment (cpurta) [inline internal/cassandra/cassandra_client.go]: "returns a new Cassandra Client interface for interacting with Cassandra"

File: internal/cassandra/cassandra_client.go [hunk 8]
```diff
@@ -9,16 +9,19 @@ import (
...
 func NewCassandraClient(session *gocql.Session) *cassandraClient {
 	return &cassandraClient{
 		session: session,
 	}
 }
 
+// GetTradesBetween Returns the transactions that occurred in a date range
```
Comment (cpurta) [inline internal/cassandra/cassandra_client.go]: "returns" instead of "Returns"

File: internal/cassandra/client.go [hunk 9]
```diff
@@ -6,6 +6,7 @@ import (
 	"github.com/cpurta/tatanka/internal/model"
 )
 
+// Client cassandra_client methods abstraction
```
Comment (cpurta) [inline internal/cassandra/client.go]: Client is the interface that provides operations on a Cassandra cluster to read and write Trades, Periods, and Simulation results.

File: internal/command/list_selectors.go [hunk 10]
```diff
@@ -5,6 +5,8 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+// ListSelectorCommand Print all available exchange options
```
Comment (cpurta) [inline internal/command/list_selectors.go]: "prints all available product ids for each supported exchange"

File: internal/command/balance.go [hunk 11]
```diff
@@ -5,6 +5,10 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+// BalanceCommand Return account Balance
+// with the option to change to another currency
+// and change config file path
```
Comment (cpurta) [inline internal/command/balance.go]: "returns the current account balances with the option to change to another currency"

File: internal/command/backfill.go [hunk 12]
```diff
@@ -5,6 +5,10 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+// BackfillCommand returns the history of trades
+// with the option to different path for config file,
+// number of days with occurred or with debug
```
Comment (cpurta) [inline internal/command/backfill.go]: "allows for historical data to be manually back filled into a configured Cassandra cluster for trading simulations"

**Review by cpurta — COMMENTED** [review]
Overall this is a great start. Most of the comments just need a bit more refinement. Also I made some comments but the private level struct do not need documentation at this time.

**Comment by spinales** [comment]
Hey @cpurta, thanks for the guides, I will have a more accurate comment next time, in the end you had to make the issue, sorry for that

**Comment by cpurta** [comment]
@DeathWofl this is looking good. It looks like there is a conflict in the `/internal/command/list_selectors.go` file. Let me know if you need some help resolving that. Once that is done it should be ready to merge.

**Comment by cpurta** [comment]
@DeathWofl, this looks great. Merging. Thanks for your help on this!

**Pull request merged**